examples/textual_inversion/textual_inversion.py

```diff
@@ -0,0 +1,547 @@
+import argparse
+import itertools
+import math
+import os
+import random
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+import torch.utils.checkpoint
+from torch.utils.data import Dataset
+
+import PIL
+from accelerate import Accelerator
+from accelerate.logging import get_logger
+from accelerate.utils import set_seed
+from diffusers import AutoencoderKL, DDPMScheduler, PNDMScheduler, StableDiffusionPipeline, UNet2DConditionModel
+from diffusers.hub_utils import init_git_repo, push_to_hub
```

Comment:
Not a huge fan of this function btw `init_git_repo` is a weird API to me. Think it'd be much cleaner to import the `Repositiory` directly from the Hub

Reply:
Also cc @anton-l 

We should not use this function in examples scripts. Note that we have this function mainly in `transformers` for the `Trainer` which handles all of the training for the user, so is by default more abstract. We however never use this function in the accelerate training scripts in Transformers, but instead work directly with the Hub repo: https://github.com/huggingface/transformers/blob/f719c0377f7f97c4bf9b6b54de209f4aad0aef4b/examples/pytorch/text-classification/run_glue_no_trainer.py#L241

I think it's important that we also do this here. Note that example scripts are supposed to be easy to tweak and easy to read. We want people to better understand the API of `huggingface_hub` to work directly with it, so let's not abstract the logic away here and add the corresponding lines.

Also @LysandreJik @sgugger what do you think?

Reply:
Agreed!